src/scry/workspace.cr

```diff
@@ -23,12 +23,12 @@ module Scry
 
     def put_file(params : DidOpenTextDocumentParams)
       file = TextDocument.new(params)
-      if file.untitled?
-        @open_files[file.filename] = {file, Completion::MethodDB.new}
-      else
+      if @dependency_graph[file.filename]?
         file_dependencies = @dependency_graph[file.filename].descendants.map &.value
         method_db = Completion::MethodDB.generate(file_dependencies)
         @open_files[file.filename] = {file, method_db}
+      else
+        @open_files[file.filename] = {file, Completion::MethodDB.new}
```

Comment:
Passing a new method db will leave the file with no method completion, if instead you generate a method_db from prelude.cr (https://github.com/crystal-lang/crystal/blob/master/src/prelude.cr) you can at least have method completion from crystal standard lib. 
You'll still need to check that that file was loaded. 

The prelude file is checked here https://github.com/crystal-lang-tools/scry/blob/c48f421418900a370670408566963a01e5fb829b/src/scry/completion/dependency_graph.cr#L73 
you can use that to set a prelude_path constant on the dependancy graph.

Reply:
Ok, no problem then, `file.untitled?` is needed, right?

So, if `graph[/src\/prelude.cr$/] ` is already checking file existence then why we are getting this? :

```
{"jsonrpc":"2.0","error":{"code":-32001,"message":"Missing hash key: \"/home/main/Projects/interactive-
crystal/mandel.cr\"","data":["/usr/lib/crystal/hash.cr:0:9 in 'fetch'","/usr/lib/crystal/hash.cr:62:5 in 
'[]'","/home/main/Projects/scry/src/scry/completion/dependency_graph.cr:32:7 in 
'[]'","/home/main/Projects/scry/src/scry/workspace.cr:29:29 in 
'put_file'","/home/main/Projects/scry/src/scry/context.cr:138:7 in 
'dispatch_notification'","/home/main/Projects/scry/src/scry/context.cr:36:7 in 
'dispatch'","/home/main/Projects/scry/src/scry.cr:23:19 in 
'start'","/home/main/Projects/scry/src/scry.cr:40:1 in '__crystal_main'","/usr/lib/crystal/crystal/main.cr:11:3 
in '_crystal_main'","/usr/lib/crystal/crystal/main.cr:112:5 in 
'main_user_code'","/usr/lib/crystal/crystal/main.cr:101:7 in 'main'","/usr/lib/crystal/crystal/main.cr:135:3 in 
'main'","__libc_start_main","_start","???"]}}
```

See specifically this line:

```
/home/main/Projects/scry/src/scry/completion/dependency_graph.cr:32:7
```

Maybe we can use `prelude_node = graph[/src\/prelude.cr$/]?` as well :sweat_smile:

Reply:
Oh, wait, I just confirmed, you're right :sweat_smile: :+1:

Reply:
@laginha87 Well,I checked my code and still getting `Missing hash key` error when I open external files. this PR changes are the only fix I have found :sweat_smile: 

But I think this PR changes do any harm. I mean, this is just checking if the file exist in the graph, right? then if exists we can do something.

Let me explain :wink: 

```cr
if @dependency_graph[file.filename]?  # If the file doesn't exist here
  file_dependencies = @dependency_graph[file.filename].descendants.map &.value # How to get something here?
  method_db = Completion::MethodDB.generate(file_dependencies)
  @open_files[file.filename] = {file, method_db}
```

Reply:
> if instead you generate a method_db from `prelude.cr` you can at least have method completion from crystal standard lib. You'll still need to check that that file was loaded.

Oh I get it!, sorry by the confusion :sweat_smile: